File: fixtures/scale/java/large/src/cryptobench/ciphers/xorcipher/XorCipherSuite.java
package cryptobench.ciphers.xorcipher;

import cryptobench.core.Cipher;
import cryptobench.core.CipherSuite;
import cryptobench.core.SuiteResult;
import cryptobench.verify.RoundTrip;

public class XorCipherSuite implements CipherSuite {
    @Override
    public String name() {
        return "xorcipher";
    }

    @Override
    public SuiteResult run() {
        Cipher cipher = new XorCipherCipher(XorCipherKey.defaultKey());
        int passed = 0;
        int failed = 0;
        long start = System.nanoTime();
        for (String sample : XorCipherVectors.samples()) {
            if (RoundTrip.check(cipher, sample)) passed++; else failed++;
        }
        return new SuiteResult(name(), passed, failed, System.nanoTime() - start);
    }
}

File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/xorcipher/XorCipherSuite.kt
package cryptobench.ciphers.xorcipher

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.roundTrip

class XorCipherSuite : CipherSuite {
    override fun name(): String = "xorcipher"

    override fun run(): SuiteResult {
        val cipher: Cipher = XorCipherCipher(XorCipherKey.default())
        var passed = 0
        var failed = 0
        val start = System.nanoTime()
        for (sample in XorCipherVectors.samples()) {
            if (roundTrip(cipher, sample)) passed++ else failed++
        }
        return SuiteResult(name(), passed, failed, System.nanoTime() - start)
    }
}
